index.js

```diff
@@ -2,59 +2,152 @@
+	var date = json.date;
+	var charName = json.characterName;
+	var error = testDateAndName(date,charName);
 
-    /*
-     Gets the score (positive and negative) for a character on a given day
-     Input:
-     {
-     "characterName" : "Jon Snow",
-     "date" : "2016-03-18T"
-     }
-     */
-    getSentimentForName: function (json, callback) {
-        //DUMMY RESPOSE, TO BE REPLACED
-        var resp = {
-            "characterName": json.characterName,
-            "date": json.date, //date of the tweets
-            "posSum": 23, //sum of the positive sentiment score on that given day
-            "negSum": 21,   //sum of the negative sentiment score on that given day
-            "posCount": 11, //count of positive tweets that day
-            "negCount": 5, //sum of negative tweets that day
-            "nullCount": 8 //sum of neutral tweets that day
-        }; //not an array, single element response
-        callback(resp);
-    },
-    /*
-     returns Analysis over a timeframe (same as above)
-     Input json:
-     {
-     "name" : "Some Name",
-     "startDate" : ISODate",
-     "endDate" : 'ISODate"
-     }
-     */
-    getSentimentForNameTimeframe: function (json, callback) {
-        //DUMMY RESPOSE, TO BE REPLACED
-        var resp = [{
-            "characterName": "Jon Snow",
-            "date": "2016-03-18T14:40:42.782Z",
-            "posSum": 23,
-            "negSum": 21,
-            "posCount": 11,
-            "negCount": 5,
-            "nullCount": 8
-        }]; //note that this in array and will usually contain more than one element
-        callback(resp);
+	//mongodb api here, then handle the response from mongodb
+	// always error in callback
+	database.getSentimentForNameTimeframe(charName,date,date,function(json){callback(json,error);});
 
-    },
+	/*
+	 //DUMMY RESPOSE, TO BE REPLACED
+	var resp = {
+		"characterName": json.characterName,
+		"date": json.date, //date of the tweets
+		"posSum": 23, //sum of the positive sentiment score on that given day
+		"negSum": 21,   //sum of the negative sentiment score on that given day
+		"posCount": 11, //count of positive tweets that day
+		"negCount": 5, //sum of negative tweets that day
+		"nullCount": 8 //sum of neutral tweets that day
+	}; //not an array, single element response
+	*/
+
+},
+/*
+ returns Analysis over a timeframe (same as above)
+ Input json:
+ {
+ "name" : "Some Name",
+ "startDate" : ISODate",
+ "endDate" : 'ISODate"
+ }
+ */
+getSentimentForNameTimeframe: function (json, callback) {
+	var name = json.name;
+	var startDate = json.startDate;
+	var endDate = json.endDate;
+	//if (testDate(startDate,charName)){console.log('Error');return;}
+	if (endDate === undefined){
+		//basiclly using getSentimentForName()
+
+	}
+	else {
+		//if(testDate(endDate,charName)){console.log('Error');return;}
+		//mongodb api here, then handle the response from mongodb
+		database.getSentimentForNameTimeframe(name,startDate,endDate,function(json){callback(json);});
```

Comment:
Where is here no error handling? You have to catch wrong user input and give back a meaningful errormessage. This is crucial in a project with different teams.